### DISCUSSION TITLE
Default WCO setting

### DISCUSSION THREAD

**Discussion by HattonLe** [fra589/grbl-Mega-5X#216] [Q&A] [answered]
Up to now I have been testing GRBL by hand using TeraTerm and the two axis I have built so far work fine for limit switch \ max movement. I then tried using UGS (Universal Gcode Sender) just to see what that GUI was like. The next day (powering up GRBL) and back to using TeraTerm I found my X axis wasn't moving as per expected. Doing a bit more digging I discovered WCO was set to non zero values (UGS had invisibly set it).
From an internet search I used "G10 L2 P1X0 Y0 Z0" to reset WCO, so now my GRBL works as it did before trying out UGS.

My questions - WCO seems to be persistent after power off - should it be? If it were to be reset to (0,0,0) on power up that would avoid my confusion above. As an aside, what is the default coordinate system on GRBL power up - is this also persistant? and where are both values held (I can't offhand see where in the code).
Thanks.
Ian

**Comment by fra589** [comment, marked as answer]
Hi Ian,

The WCO offset is a combination of global offset (G92) and one of the 6 coordinate systems (G54 to G59).
G54 to 59 coordinates system are persistent after power off, G92 is not...
See http://linuxcnc.org/docs/html/gcode/coordinates.html#cha:coordinate-system for more informations.

At power on or reset, the active coordinate system is G54.
You can automatically reset the current coordinate system at power on by configuring the Grbl startup block with the reset GCode command:
$N0=G10L2P0X0Y0Z0
See : https://github.com/fra589/grbl-Mega-5X/wiki/Grbl-Commands#n---view-startup-blocks

@++;
Gauthier.
